### ISSUE TITLE
try groupby instead of where

### ISSUE THREAD

**Issue by rabernat** [RossOng/RCES_FinalProject#1]
I'm slowly reviewing the final projects. Yours is really excellent! I'm especially pleased that your binder really works. 

Just a tiny technical suggestion. The way you are currently getting the seasonal mean is

```python
sst_jfm_plot = nat_sst_anom.where((nat_sst_anom['time.month'] <= 3)).mean(dim='time')
```

I encourage you to not use `.where` for selecting data like this. It's slightly inefficient to do it this way, because you are techincally filling in `NaN`s everywhere where your condition is not met, then iterating over the whole array. Basically it creates extra work for your computer.

Instead, consider grouping by season.
```python
sst_seasonal_mean = nat_sst_anom.groupby('time.season').mean(dim='time')
sst_seasonal_mean.sel(season='DJF').plot()
```
This is both faster and easier to write / read.

**Comment by RossOng** [comment]
Hi Ryan, thanks for the feedback. Really appreciate the time you took to go through my project.

I tried to fix the code and typed this to get JFM seasons specifically:

`sst_seasonal_mean = nat_sst_anom.groupby('time.season').mean(dim='time')
sst_seasonal_mean.sel(season='JFM').plot()`

However I got an error message. It seems like 'time.season' only works for DJF, MAM, JJA, SON seasons. I could use this for when I get my JJA season. But for JFM, do you have any suggestion on how I can use groupby to select specific seasons aside from the 4 seasons above?

**Comment by rabernat** [comment]
Good point. You would have to define a custom grouper function, as in this
example. Probably not worth the trouble.
https://stackoverflow.com/questions/25647546/pandas-efficient-groupby-season-for-every-year

**Comment by RossOng** [comment]
Good to know an alternative. I will continue working on this for my research, I'll be sure to share thoughts if I find a better way. Thank you!
